numpy/core/tests/test_item_selection.py

```diff
@@ -68,6 +68,21 @@ def test_unicode_mode(self):
         k = b'\xc3\xa4'.decode("UTF8")
         assert_raises(ValueError, d.take, 5, mode=k)
 
+    def test_partition(self):
+        # In reference to github issue #6530
+        a = np.array([0, 2, 4, 6, 8, 10])
+        a = a.partition(np.array([], dtype=np.int16))
+
+        assert_equal(a, None)
```

Comment:
The above two lines don't make much sense -- `.partition` acts in-place and always returns `None`. It works to ensure that it isn't raising an error, but a better way to write the test would be:
```
a_original = np.array([0, 2, 4, 6, 8, 10])
a = a_original.copy()
# An empty partition should be a successful no-op
a.partition(np.array([], dtype=np.int16))
assert_equal(a, a_original)
```